File: cbn/cbn/custom/quality_inspection.py
# Copyright (c) 2024, DAS and Contributors
# License: GNU General Public License v3. See license.txt

import frappe

def set_job_card_bm(self, method=None):
    if self.reference_type != "Job Card":
        return
    
    self.custom_batch = frappe.get_value(self.reference_type, self.reference_name, "custom_batch")

@frappe.whitelist()
def get_job_card(job_card):
    jc = frappe.qb.DocType("Job Card")
    wo = frappe.qb.DocType("Work Order")

    query = (
        frappe.qb.from_(jc)
        .inner_join(wo)
        .on(jc.work_order == wo.name)
        .select(
            jc.actual_start_date,
            jc.actual_end_date,
            wo.qty,
            wo.custom_line_produksi,
        )
        .where(jc.name == job_card)
        .limit(1)
    )

    data = query.run(as_dict=True)

    return data[0] if data else {}

File: cbn/cbn/custom/quality_inspection.js
// Copyright (c) 2024, DAS and contributors
// For license information, please see license.txt

frappe.ui.form.on("Quality Inspection", {
    reference_name(frm){
        if(frm.doc.reference_type == "Job Card" && frm.doc.reference_name){
            frappe.call({
                method: "cbn.cbn.custom.quality_inspection.get_job_card",
                args: {
                    job_card: frm.doc.reference_name
                },
                callback: (r) => {
                    let data = r.message
                    if (data) {
                        frm.set_value("custom_batch_size", r.qty);
                        frm.set_value("custom_line_produksi", r.custom_line_produksi);
                        frm.set_value("custom_start_filling", r.actual_start_date);
                        frm.set_value("custom_end_filling", r.actual_end_date);
                        frm.set_value("custom_start_packing", r.actual_start_date);
                        frm.set_value("custom_end_filling", r.actual_end_date);
                    }
                }
            })
        }
    }
})